src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ClientConnectionManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Claims;
+using System.Web;
+using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Hosting;
+using Microsoft.Azure.SignalR.Protocol;
+using Microsoft.Owin;
+
+namespace Microsoft.Azure.SignalR.AspNet
+{
+    internal class ClientConnectionManager : IClientConnectionManager
+    {
+        private static readonly ClaimsPrincipal EmptyPrincipal = new ClaimsPrincipal(new ClaimsIdentity());
+
+        private static readonly string[] SystemClaims =
+        {
+            "aud", // Audience claim, used by service to make sure token is matched with target resource.
+            "exp", // Expiration time claims. A token is valid only before its expiration time.
+            "iat", // Issued At claim. Added by default. It is not validated by service.
+            "nbf"  // Not Before claim. Added by default. It is not validated by service.
+        };
+
+        private readonly PersistentConnection _dispatcher;
+
+        public ClientConnectionManager(IDependencyResolver resolver)
+        {
+            _dispatcher = resolver.Resolve<PersistentConnection>() ?? throw new ArgumentNullException(nameof(resolver));
+        }
+
+        public AzureTransport CreateConnection(OpenConnectionMessage message)
+        {
+            var connectionId = message.ConnectionId;
+
+            var responseStream = new MemoryStream();
+            var hostContext = GetHostContext(message, responseStream);
+
+            if (_dispatcher.Authorize(hostContext.Request))
+            {
+                // ProcessRequest checks if the connectionToken matches "{connectionid}:{userName}" format with context.User
+                _ = _dispatcher.ProcessRequest(hostContext);
+
+                // TODO: check for errors written to the response
+                if (hostContext.Response.StatusCode != 200)
+                {
+                    Debug.Fail("Response StatusCode is " + hostContext.Response.StatusCode);
+                    var errorResponse = GetMemoryString(responseStream);
+                    throw new InvalidOperationException(errorResponse);
+                }
+                return (AzureTransport)hostContext.Environment[Constants.Context.AzureSignalRTransportKey];
+            }
+
+            // This happens when hub is not found
+            Debug.Fail("Unauthorized");
+            throw new InvalidOperationException("Unable to authorize request");
```

Comment:
Will this end up sending a close?

Reply:
Yes, in https://github.com/Azure/azure-signalr/pull/159/files/74cd48519753460f87f4e030b18068447ef66576#diff-5e3f883b4759d1fe0925a6e812dc026bR122